lib/libzfs/libzfs_pool.c

```diff
@@ -3854,6 +3854,10 @@ zpool_history_unpack(char *buf, uint64_t bytes_read, uint64_t *leftover,
 		if (ISP2(*numrecords + 1)) {
 			*records = realloc(*records,
 			    *numrecords * 2 * sizeof (nvlist_t *));
+			if (*records == NULL) {
```

Comment:
As i understand the code, if there is any error (for ex. ENOMEM), then `./cmd/spool/zpool_main.c` will give this error to user as exit code, but i may misunderstand if.